src/Core/Liz.Core/Extract/ExtractLicenses.cs

```diff
@@ -94,8 +111,37 @@ private async Task<IEnumerable<PackageReference>> GetPackageReferencesForProject
         }
         catch (Exception ex)
         {
-            throw new InvalidOperationException(
-                $"Could not determine package-references for project '{project.Name} ({project.File})'", ex);
+            throw new GetPackageReferencesFailedException(project, ex);
         }
     }
+
+    private async Task EnrichWithLicenseInformationAsync(
+        IEnumerable<PackageReference> packageReferences)
+    {
+        _logger.LogDebug("Trying to get license information for package(s)...");
+
+        foreach (var packageReference in packageReferences)
+            await EnrichWithLicenseInformationForPackageReferenceAsync(packageReference);
+    }
+
+    private async Task EnrichWithLicenseInformationForPackageReferenceAsync(PackageReference packageReference)
+    {
+        try
+        {
+            _logger.LogDebug($"Trying to get license information for {packageReference}...");
+
+            await _enrichPackageReferenceWithLicenseInformation.GetFromPackageReferenceAsync(packageReference);
```

Comment:
`GetFromPackageReferenceAsync` is not a great name for that method, change it to something like `EnrichPackageReferenceAsync`